cli/src/validate.rs

```diff
@@ -62,44 +61,44 @@ pub async fn validate(junit_paths: Vec<String>, show_warnings: bool) -> anyhow::
     }
 }
 
-fn parse_file_sets(file_sets: Vec<FileSet>) -> anyhow::Result<JunitFileToReportAndErrors> {
-    file_sets.iter().try_fold(
+fn parse_file_sets(file_sets: Vec<FileSet>) -> JunitFileToReportAndErrors {
+    file_sets.iter().flat_map(|file_set| &file_set.files).fold(
         JunitFileToReportAndErrors::new(),
-        |mut file_sets_parse_results, file_set| -> anyhow::Result<JunitFileToReportAndErrors> {
-            let file_set_parse_results = file_set.files.iter().try_fold(
-                JunitFileToReportAndErrors::new(),
-                |mut file_set_parse_results,
-                 bundled_file|
-                 -> anyhow::Result<JunitFileToReportAndErrors> {
-                    let path = std::path::Path::new(&bundled_file.original_path);
-                    let file = std::fs::File::open(path)?;
-                    let file_buf_reader = BufReader::new(file);
-                    let mut junit_parser = JunitParser::new();
-                    junit_parser.parse(file_buf_reader).context(format!(
-                        "Encountered unrecoverable error while parsing file: {}",
-                        bundled_file.original_path_rel
-                    ))?;
-
-                    let mut cur_file_parse_results = file_set_parse_results
-                        .get(&bundled_file.original_path_rel)
-                        .cloned()
-                        .unwrap_or((Vec::new(), Vec::new()));
-
-                    cur_file_parse_results.1.extend(junit_parser.errors());
-                    cur_file_parse_results.0.extend(junit_parser.into_reports());
-
-                    file_set_parse_results.insert(
+        |mut parse_results, bundled_file| -> JunitFileToReportAndErrors {
+            let path = std::path::Path::new(&bundled_file.original_path);
+            let file = match std::fs::File::open(path) {
+                Ok(file) => file,
+                Err(e) => {
+                    parse_results.insert(
                         bundled_file.original_path_rel.clone(),
-                        cur_file_parse_results,
+                        (Err(anyhow::anyhow!(e)), Vec::new()),
                     );
+                    return parse_results;
+                }
+            };
 
-                    Ok(file_set_parse_results)
-                },
-            )?;
+            let file_buf_reader = BufReader::new(file);
+            let mut junit_parser = JunitParser::new();
+            match junit_parser.parse(file_buf_reader) {
+                Err(e) => {
+                    parse_results.insert(
+                        bundled_file.original_path_rel.clone(),
+                        (Err(anyhow::anyhow!(e)), Vec::new()),
+                    );
+                    return parse_results;
+                }
+                _ => (),
+            };
```

Comment:
nit: it's better to use `if let` for cases like this where we don't care about matching other variants

```suggestion
            if let Err(e) = junit_parser.parse(file_buf_reader) {
                parse_results.insert(
                    bundled_file.original_path_rel.clone(),
                    (Err(anyhow::anyhow!(e)), Vec::new()),
                );
                return parse_results;
            }
```